features/verblijfplaats.feature

```diff
@@ -0,0 +1,140 @@
+# language: nl
+
+@proxy @post-assert
+Functionaliteit: leveren van een verblijfplaats
+
+  Rule: Een verblijfplaats is onbekend wanneer land (13.10) de waarde '0000' heeft
+    - veld 'type' krijgt de waarde 'VerblijfplaatsOnbekend' 
+    - veld 'verblijfplaatsOnbekend' wordt opgenomen met de boolean waarde true
+
+    Scenario: land onbekend 
+      Gegeven het systeem heeft een persoon met de volgende gegevens
+      | naam                | waarde    |
+      | burgerservicenummer | 555550001 |
+      En de persoon heeft de volgende 'verblijfplaats' gegevens
+      | naam                                | waarde   |
+      | gemeenteVanInschrijving (09.10)     | 1999     |
+      | datumInschrijvingInGemeente (09.20) | 20220128 |
+      | land (13.10)                        | 0000     |
+      | datumAanvangAdresBuitenland (13.20) | 20220128 |
+      Als personen wordt gezocht met de volgende parameters
+      | naam                | waarde                             |
+      | type                | RaadpleegMetBurgerservicenummer    |
+      | burgerservicenummer | 555550001                          |
+      | fields              | burgerservicenummer,verblijfplaats |
+      Dan heeft de persoon met burgerservicenummer '555550001' de volgende 'verblijfplaats' gegevens
+      | naam                              | waarde                 |
+      | type                              | VerblijfplaatsOnbekend |
+      | verblijfplaatsOnbekend            | true                   |
+      | gemeenteVanInschrijving           | 1999                   |
+      | datumInschrijvingInGemeente.type  | Datum                  |
+      | datumInschrijvingInGemeente.datum | 2022-01-28             |
+
+  Rule: Een verblijfplaats is een adres of locatie in het buitenland wanneer land (13.10) een waarde ongelijk aan '0000' (onbekend) heeft
+    - veld 'type' krijgt de waarde 'VerblijfplaatsBuitenland'
+
+    Abstract Scenario: verblijfplaats buitenland
```

Comment:
Zoals de scenario's zijn geschreven, worden er eigenlijk meer dingen tegelijk geïllustreerd, maar ook niet alles. voorbeeld bij verblijfplaats buitenland scenario:
- als land is ongelijk aan '0000', dan is type VerblijfBuitenland
- datumAanvangAdresBuitenland wordt gemapt naar datumVan
- mappen van adresregel3 wordt niet geïllustreerd
- omzetten van datum velden. Hoort dit eigenlijk niet in de datum feature? Persoonlijk zou ik het niet doen omdat je hiermee op meerdere plekken iets over verblijfplaats gegevens illustreert.

Ik denk dat het gebruiken van meerdere scenario's verblijfplaats buitenland gegeven beter illustreert dan één scenario. Hieronder mijn poging:

```
    Abstract Scenario: waarde land (13.10) is ongelijk aan '0000'
      Gegeven het systeem heeft een persoon met de volgende gegevens
      | naam                | waarde    |
      | burgerservicenummer | 555550001 |
      En de persoon heeft de volgende 'verblijfplaats' gegevens
      | naam         | waarde |
      | land (13.10) | <land> |
      Als personen wordt gezocht met de volgende parameters
      | naam                | waarde                             |
      | type                | RaadpleegMetBurgerservicenummer    |
      | burgerservicenummer | 555550001                          |
      | fields              | burgerservicenummer,verblijfplaats |
      Dan heeft de persoon met burgerservicenummer '555550001' de volgende 'verblijfplaats' gegevens
      | naam      | waarde                   |
      | type      | VerblijfplaatsBuitenland |
      | land.code | <land>                   |

      Voorbeelden:
      | land |
      | 6014 |
      | 9999 |

    Abstract Scenario: verblijfplaats buitenland. Overnemen veld: <gba naam>
      Gegeven het systeem heeft een persoon met de volgende gegevens
      | naam                | waarde    |
      | burgerservicenummer | 555550001 |
      En de persoon heeft de volgende 'verblijfplaats' gegevens
      | naam         | waarde   |
      | land (13.10) | 6014     |
      | <gba naam>   | <waarde> |
      Als personen wordt gezocht met de volgende parameters
      | naam                | waarde                                    |
      | type                | RaadpleegMetBurgerservicenummer           |
      | burgerservicenummer | 555550001                                 |
      | fields              | burgerservicenummer,verblijfplaats.<naam> |
      Dan heeft de persoon met burgerservicenummer '555550001' de volgende 'verblijfplaats' gegevens
      | naam   | waarde                   |
      | type   | VerblijfplaatsBuitenland |
      | <naam> | <waarde>                 |

      Voorbeelden:
      | gba naam                             | naam                                 | waarde                             |
      | adresregel1 (13.30)                  | adresregel1                          | 1600 Pennsylvania Avenue NW        |
      | adresregel2 (13.40)                  | adresregel2                          | Washington, DC 20500               |
      | adresregel3 (13.50)                  | adresregel3                          | Selangor                           |
      | gemeente van inschrijving (09.10)    | gemeenteVanInschrijving.code         | 1999                               |
      | gemeenteVanInschrijving.omschrijving | gemeenteVanInschrijving.omschrijving | Registratie Niet Ingezetenen (RNI) |
      | land.omschrijving                    | land.omschrijving                    | Verenigde Staten van Amerika       |

    Abstract Scenario: verblijfplaats buitenland. Omzetten datum veld: <gba naam>
      Gegeven het systeem heeft een persoon met de volgende gegevens
      | naam                | waarde    |
      | burgerservicenummer | 555550001 |
      En de persoon heeft de volgende 'verblijfplaats' gegevens
      | naam         | waarde       |
      | land (13.10) | 6014         |
      | <gba naam>   | <gba waarde> |
      Als personen wordt gezocht met de volgende parameters
      | naam                | waarde                                    |
      | type                | RaadpleegMetBurgerservicenummer           |
      | burgerservicenummer | 555550001                                 |
      | fields              | burgerservicenummer,verblijfplaats.<naam> |
      Dan heeft de persoon met burgerservicenummer '555550001' de volgende 'verblijfplaats' gegevens
      | naam         | waarde                   |
      | type         | VerblijfplaatsBuitenland |
      | <naam>.type  | Datum                    |
      | <naam>.datum | <waarde>                 |

      Voorbeelden:
      | gba naam                            | gba waarde | naam                        | waarde     |
      | datumInschrijvingInGemeente (09.20) | 20020701   | datumInschrijvingInGemeente | 2002-07-01 |
      | datumAanvangAdresBuitenland (13.20) | 20220128   | datumVan                    | 2022-01-28 |

```

Reply:
@MelvLee ik heb dit scenario en ook de andere scenario's beperkt tot alleen de velden die relevant zijn voor het bepalen en tonen van het type.

Zullen we andere uitbreiding van het feature, zoals mapping en bepalen afgeleide gegevens hier nu buiten laten en op een later moment (een volgende PR) oppakken? Een deel daarvan staat al in features adres en woonplaats, maar moeten worden aangepast en uitgebreid voor de v2.0.0 aanpassingen